packages/playground-examples/copy/ja/TypeScript/Meta-Types/Conditional Types.ts

```diff
@@ -0,0 +1,109 @@
+// 条件付き型はTypeScriptの型システムに
+// 簡単なロジックを組み込む方法を提供します。
+// これは非常に高度な機能のため、
+// 日々の開発において使わないことも十分に可能です。
+
+// 条件付き型は以下のような形です:
+//
+//   A extends B ? C : D
+//
+// 条件部はある型がある式を拡張しているかを定義し、
+// 条件を満たす場合にどの型を返すかを定義します。
+
+// いくつかの例を見てみましょう。
+// ここでは簡潔さのためにジェネリクスに単一の文字を使用していきます。
+// これは任意ですが、
+// 60文字に制限すると画面に収まり難くなります。
+
+type Cat = { meows: true };
+type Dog = { barks: true };
+type Cheetah = { meows: true; fast: true };
+type Wolf = { barks: true; howls: true };
+
+// 以下のように吠える(barks)動物の型だけを
+// 抜き出す条件付き型を作れます。
+
+type ExtractDogish<A> = A extends { barks: true } ? A : never;
+
+// 次に、ExtractDogishで包んだ型を作成します:
+
+// Cat型は吠えないので、neverが返ります。
+type NeverCat = ExtractDogish<Cat>;
+// Wolf型は吠えるので、Wolf型が返ります。
+type Wolfish = ExtractDogish<Wolf>;
+
+// これは多くの型を含む交差型を扱って、
+// その交差型に含まれる型の数を
+// 絞りたい際に有用です:
+
+type Animals = Cat | Dog | Cheetah | Wolf;
+
+// 交差型にExtractDogish型を適用するのは、
+// その交差型に含まれるそれぞれの型に
+// 条件を当てはめるのと同じです:
+
+type Dogish = ExtractDogish<Animals>;
+
+// = ExtractDogish<Cat> | ExtractDogish<Dog> |
+//   ExtractDogish<Cheetah> | ExtractDogish<Wolf>
+//
+// = never | Dog | never | Wolf
+//
+// = Dog | Wolf (example:unknown-and-never を参照)
+
+// これは交差型のそれぞれの型に型が割り当てられるため、
+// 分配的条件付き型と呼ばれます。
```

Comment:
[nits]
型が多くて、何を表しているかがわかりづらいと思いました。↓ではいかがでしょうか？

```suggestion
// これは共用型のそれぞれの型に対して条件型が割り当てられるため、
// 分配条件型と呼ばれます。
```

[Q]

こちらも、`分配条件型` とかでしょうか？
(distributive conditional type は、web上では `union型の分配` などの表現しか見つからず、、、)

Reply:
Conditional typesが英語のままなので、おとなしく、英語のままにしましょう
```suggestion
// これは共用型のそれぞれの型にConditional Typesが割り当てられるため、
// Distributive Conditional Typesと呼ばれます。
```